docs/auto_aug/augmentations.rst

```diff
@@ -0,0 +1,259 @@
+
+    rotate_aug_60 = rotate_aug.augmentation(mag_range=(0, 60), randomly_negate=False)
+
+
+Decorator
+---------
+
+.. autodecorator:: nvidia.dali.auto_aug.core.augmentation
+
+.. currentmodule:: nvidia.dali.auto_aug._augmentation
+
+.. class:: nvidia.dali.auto_aug.core._augmentation.Augmentation
+
+    The result of decorating a function with :py:func:`@augmentation <nvidia.dali.auto_aug.core.augmentation>`
+    is an instance of class :meth:`~nvidia.dali.auto_aug.core._augmentation.Augmentation`.
+    The class should not be instantiated directly, it needs to be created with the decorator.
+
+    Once obtained, those objects become callables that can be used to speicfy a policy for
+    :ref:`AutoAugment <auto_augment>`, :ref:`RandAugment <rand_augment>`
+    or :ref:`TrivialAugment <trivial_augment>`.
+
+    .. method:: def augmentation(self, mag_range, randomly_negate, mag_to_param, param_device, name) -> Augmentation
+
+        You can call this method to create new ``Augmentation`` instance based on an existing one,
+        with the paramenters adjusted. All parameters are optional - those that were specifed
+        replace the ones that were previouslly passed to
+        :py:func:`@augmentation <nvidia.dali.auto_aug.core.augmentation>`.
+
+        :param mag_range: optional, see :py:func:`@augmentation <nvidia.dali.auto_aug.core.augmentation>`.
+        :param randomly_negate: optional, see :py:func:`@augmentation <nvidia.dali.auto_aug.core.augmentation>`.
+        :param mag_to_param: optional, see :py:func:`@augmentation <nvidia.dali.auto_aug.core.augmentation>`.
+        :param param_device: optional, see :py:func:`@augmentation <nvidia.dali.auto_aug.core.augmentation>`.
+        :param name: optional, see :py:func:`@augmentation <nvidia.dali.auto_aug.core.augmentation>`.
+
+
+Augmentations
+-------------
+
+Here is a list of callable :meth:`~nvidia.dali.auto_aug.core._augmentation.Augmentation` instances
+defined by DALI. Note that the ``mag_to_param``, ``param_device`` and ``name`` parameters were
+ommited from the :py:func:`@augmentation <nvidia.dali.auto_aug.core.augmentation>` decorator listing
+for simplicity.
+
+To adjust the range of parameter, use the ``augmentation`` method
+on the existing :meth:`~nvidia.dali.auto_aug.core._augmentation.Augmentation` instance listed below,
+for example::
+
+    # Create a steeper sheer operation based on existing one
+    steep_shear_x = shear_x.augmentation(mag_range=(0, 0.5), name="steep_shear_x")
+
+
+.. currentmodule:: nvidia.dali.auto_aug.augmentations
+
+.. function:: shear_x(sample, *, magnitude_bin=None, num_magnitude_bins=None, **kwargs)
+
+Applies :meth:`nvidia.dali.fn.transforms.shear` with ``shear_x`` factor using
+:meth:`nvidia.dali.fn.warp_affine`.
+
+
+.. code-block::
+
+    @augmentation(mag_range=(0, 0.3), randomly_negate=True, ...)
+    def shear_x(sample, shear, fill_value=128, interp_type=None)
```

Comment:
Maybe it can be combined with `shear_y`?

Reply:
not sure if I understand.

Reply:
I mean the docs for `shear_x` and `shear_y` is very repetitive and I wonder if these two ops can be somehow combined

Reply:
I won't be doing it, I plan as a next step to move the docs to py and generate them with `.. autofunction::`. In that case it would be harder to combine.

Reply:
Just an idea